[ADD] Website: creating building blocks documentation (#7800)
odoo/documentation: content/applications/websites/website/web_design/building_blocks.rst

:guilabel:`Show/Hide on Mobile` icons.

.. image:: building_blocks/show-hide-on-mobile.png
   :alt: Click the "show/hide on mobile" icons to show or hide some elements on mobile.

You can decide to show/hide with :guilabel:`No condition` or
:guilabel:`Conditionally`, depending on :guilabel:`Country` (based on Geo IP), :guilabel:`Languages`
(for multi-language websites), :guilabel:`UTM Campaign`, :guilabel:`UTM Medium`,
:guilabel:`UTM Source`, or :guilabel:`Users`.

The elements hidden on mobile/desktop view appear in the :guilabel:`Invisible Elements` section at
the bottom of the configurator. Click on a specific element to make it re-appear on screen.

.. tip::
   Click the phone icon at the top of the configurator to see how your website renders on mobile.

   .. image:: building_blocks/phone-icon.png
      :alt: Click the phone icon to view how your website renders on mobile.

Saving a custom building block
==============================

You can save your favorite custom building blocks for future use on other site pages. Select the
custom building block you want to save, navigate to the :guilabel:`Customize` tab, and click the

> floppy disk icon to store it for later application.

Reviewer: floppy disk icon to store it.

===============
Building blocks
===============

It is possible to add and design your content without coding by dragging and dropping building
blocks from the website builder onto your web pages. Odoo classifies them into four categories:
:guilabel:`Structure`, :guilabel:`Features`, :guilabel:`Dynamic Content`, and :guilabel:`Inner

> Content`.

Author: One page is being created per building block category and the link will be added for each page :)

.. seealso::
   - `Tutorial: Design your first webpage <https://www.odoo.com/slides/slide/design-your-first-web-page-1667?fullscreen=1>`_
   - `Odoo Website <https://www.odoo.com/app/website>`_

>    - :doc:`Website themes <themes>`

Author: I would specify the Website themes link is part of the documentation. 
--> Should we discuss it with everyone to agree on a single way of doing things?

The :guilabel:`Grid` layout allows you to reposition and resize elements, such as images or text, by
dragging and dropping them to the desired place.

.. image:: building_blocks/grid-layout.png
   :alt: When the grid layout is selected, choose an image and drag and drop it where needed.

.. tip::
   Position images behind the text by using the above/below icons.

   .. image:: building_blocks/superimpose-images-to-text.png
      :alt: Use the above/below icons to superimpose images to text.

Cols
----

The :guilabel:`Cols` feature is available for some blocks like :guilabel:`Columns` or
:guilabel:`Image - Text` blocks. By choosing the :guilabel:`Cols` :guilabel:`Layout`, you can
determine the number of elements per line within the block.

.. image:: building_blocks/elements-per-line.png
   :alt: Set the number of elements per line in the block.

.. note::
   In the mobile view, only one element per column is used by default for better visibility.
   However, you can adjust this setting without impacting the desktop view.

Duplicating a building block
============================

You can replicate a building block by clicking on the duplicate icon. Once duplicated, the new block

> appears on your website beneath the original one.

Author: I prefer to keep it, as it clarifies how it works.

.. image:: building_blocks/duplicate-container.png
   :alt: Replicating a building block

Re-ordering building blocks
===========================

> To re-order the building blocks on your site, select the block you want to move and click the up

Reviewer 2: ```suggestion
To reorder building blocks, select the block you want to move and click the up
```

arrow to move it before the previous section or click the down arrow to move it after the following
section.

You can also use the drag-and-drop icon to move your block manually.

.. image:: building_blocks/re-ordering-blocks.png

>    :alt: Re-order building blocks by using the up and down arrow icons.

Reviewer 2: ```suggestion
   :alt: Reorder building blocks by using the up and down arrow icons.
```

> Show/Hide on mobile

Reviewer 2: ```suggestion
Visibility of building blocks per device type or condition
```
I would change the heading to speak about all visibility options.

Author: I do not really like the title you suggested; it's too long for me. If you can live with it, I would prefer to keep the title I wrote.

Author: or maybe to speak about "visibility on mobile"?

Author: updated as discussed live

===================

> You can hide specific elements of your website for mobile devices. To do so, select the element

Reviewer 2: ```suggestion
You can hide building blocks for users on mobile devices or desktops. To do so, select the element
```

to hide, and in the :guilabel:`Customize` tab, scroll down to :guilabel:`Visibility`, and click the

> :guilabel:`Show/Hide on Mobile` icons.

Reviewer 2: ```suggestion
:guilabel:`Show/Hide on Mobile` or the :guilabel:`Show/Hide on Desktop` icon.
```

.. image:: building_blocks/show-hide-on-mobile.png
   :alt: Click the "show/hide on mobile" icons to show or hide some elements on mobile.

> You can decide to show/hide with :guilabel:`No condition` or
> :guilabel:`Conditionally`, depending on :guilabel:`Country` (based on Geo IP), :guilabel:`Languages`
> (for multi-language websites), :guilabel:`UTM Campaign`, :guilabel:`UTM Medium`,
> :guilabel:`UTM Source`, or :guilabel:`Users`.

Reviewer 2: ```suggestion
You can also show or hide building blocks depending on specific conditions.
To do so, click :guilabel:`No condition` and select :guilabel:`Conditionally` instead. Next, under one of the available,  conditions, select if the building block should be :guilabel:`Visible for` or :guilabel:`Hidden for` specific records.
```
The available conditions depend on your DB, not everyone uses multiple languages or UTM campaigns. I think describing the general process is more interesting, especially if you don't describe what are the options.

> The elements hidden on mobile/desktop view appear in the :guilabel:`Invisible Elements` section at
> the bottom of the configurator. Click on a specific element to make it re-appear on screen.

Reviewer 2: ```suggestion
Some elements can become hidden from your current view. To make a building block visible again, go to the :guilabel:`Invisible Elements` section at the bottom of the configurator and select a building block.
```

.. tip::

>    Click the phone icon at the top of the configurator to see how your website looks on mobile.

Reviewer 2: ```suggestion
   Click the phone icon at the top of the configurator to preview how your website would look on a mobile device.
```

   .. image:: building_blocks/phone-icon.png
      :alt: Click the phone icon to view how your website renders on mobile.

Saving a custom building block
==============================

> You can save your favorite custom building blocks on other site pages for future use. Select the
> custom building block you want to save, navigate to the :guilabel:`Customize` tab, and click the
> floppy disk icon to store it to use it later.

Reviewer 2: ```suggestion
You can create custom building blocks that you can reuse elsewhere. To do so, select
the building block you want to save, navigate to the :guilabel:`Customize` tab, and click the
floppy disk icon to save it.
```

.. image:: building_blocks/saving-custom-block.png
   :alt: Saving a custom block to use it elsewhere

> Your custom building blocks are available in the :guilabel:`Custom` section of the
> :guilabel:`Blocks` tab. Click on the pen icon to edit their name and easily find them again.

Reviewer 2: ```suggestion
Your custom building blocks are available in the :guilabel:`Custom` section of the
:guilabel:`Blocks` tab. You can click the pen icon to edit their name.
```

Cols
----

Choosing the :guilabel:`Cols` layout allows you to determine the number of elements per line within
the block.

.. image:: building_blocks/elements-per-line.png
   :alt: Set the number of elements per line in the block.

.. note::
   On mobile devices, only one element per column is used by default for better visibility. You can
   adjust this value by using the :guilabel:`Mobile Preview` mode.

Duplicating a building block
============================

You can duplicate a building block by clicking on the duplicate icon. Once duplicated, the new block
appears on your website beneath the original one.

.. image:: building_blocks/duplicate-container.png
   :alt: Replicating a building block

Reordering building blocks
==========================

To reorder the building blocks, select the block you want to move and click the up arrow to move it
before the previous section, or click the down arrow to move it after the following section.

You can also use the drag-and-drop icon to move your block manually.

.. image:: building_blocks/re-ordering-blocks.png
   :alt: Reorder building blocks by using the up and down arrow icons.

Visibility
==========

> Visibility on mobile

Author: updated as discussed live

--------------------

You can hide specific elements of your website for mobile devices. To do so, select the element
to hide, and in the :guilabel:`Customize` tab, scroll down to :guilabel:`Visibility`, and click the
:guilabel:`Show/Hide on Mobile` or the :guilabel:`Show/Hide on Desktop` icon.

.. image:: building_blocks/show-hide-on-mobile.png
   :alt: Click the "show/hide on mobile" icons to show or hide some elements on mobile.

.. tip::
   Click the phone icon at the top of the configurator to preview your website appearance on a
   mobile device.

   .. image:: building_blocks/phone-icon.png
      :alt: Click the phone icon to view how your website renders on mobile.

Conditional visibility
----------------------

> You can also conditionally render visible or hidden for certain building blocks. To do so, select
> the element you want to show/hide, go to :guilabel:`Visibility`, and click :guilabel:`Conditionally`.
> Then, choose the conditions to apply based on the :guilabel:`Country` (based on Geo IP),
> :guilabel:`Languages` (for multi-language websites), :guilabel:`UTM Campaign`,
> :guilabel:`UTM Medium`, :guilabel:`UTM Source`, or :guilabel:`Users`. The options available depend
> on your database's settings.

Reviewer 2: ```suggestion
You can also hide or show building blocks using other conditions. To do so, select
an element, go to :guilabel:`Visibility`, click :guilabel:`No condition`, and select :guilabel:`Conditionally` instead.
Then, configure the condition(s) to apply by selecting :guilabel:`Visible for` or :guilabel:`Hidden for` and which  :guilabel:`Records` will be impacted.
```
- The first sentence isn't clear.
- By default, you don't see ***Conditionally*** but ***No condition***
- You don't choose conditions based on _xxx_ (a condition). For a condition, you have to choose: which records the condition applies to and if the block should be shown or hidden.
